core/domain/html_validation_service.py

```diff
@@ -749,3 +753,51 @@ def _validate_customization_args_in_tag(tag):
                             yield err_msg
     except Exception as e:
         yield str(e)
+
+
+def add_dimensions_to_noninteractive_image_tag(exp_id, html_string):
+    """Adds dimensions to all oppia-noninteractive-image tags.
+
+    Args:
+        html_string: str. HTML string in which the dimensions is to be
+            added.
+        exp_id: str. Exploration id.
+
+    Returns:
+        str. Updated HTML string with the dimensions for all
+            oppia-noninteractive-image tags.
+    """
+    soup = bs4.BeautifulSoup(html_string.encode('utf-8'), 'html.parser')
+    for image in soup.findAll(name='oppia-noninteractive-image'):
+        filename = unescape_html(image['filepath-with-value'])
```

Comment:
This doesn't seem symmetric with the re-encoding two lines below, and the use of `filename[1:-1]` is a hack. Look at how customization args are parsed in other functions in this file.

Reply:
Done